docker/docker_client.go

```diff
@@ -412,7 +415,13 @@ func getAuth(ctx *types.SystemContext, registry string) (string, string, error)
 	return "", "", nil
 }
 
-func (c *dockerClient) ping() error {
+// ensurePropertiesAreDetected detects various properties of the registry.
+// See the dockerClient documentation for members which are affected by this.
+func (c *dockerClient) ensurePropertiesAreDetected() error {
```

Comment:
too long even if such expressive. `GatherProperties | GetRegistryProperties` or such would be better to me (and I'm not a native english speaker/writer, I just know Golang doesn't need such expressiveness according to the official doc I couldn't find a link for :smile: )